### PR TITLE
Added TestResultEventListener

### PR DESCRIPTION
**Pull request by nameisaravind**
TestResultEventListener is added. Whenever test result collection updated with new performance test-result document , the performance collection gets updated with new performance document.

### CODE REVIEW

**Comment by nireeshT** [comment]
@nameisaravind  There are un-related commits for this PR. Follow below steps and resubmit PR 
1) Close this PR
2) Delete your fork and refork from Hygieia/master
3) Submit PR with new changes. That should include commits related to your change.

File: core/src/main/java/com/capitalone/dashboard/event/TestResultEventListener.java [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+    public void onAfterSave(AfterSaveEvent<TestResult> event) {
+        TestResult testResult = event.getSource();
+        LOGGER.info("TestResult afterSave event is triggered");
+
+        // Ignore anything other than performance tests
+        if (!TestSuiteType.Performance.equals(testResult.getType())) {
+            return;
+        }
+        for (Object testCapabilityObj : testResult.getTestCapabilities().toArray()){
+            TestCapability testCapability = (TestCapability) testCapabilityObj;
```
Comment (rvema) [inline core/src/main/java/com/capitalone/dashboard/event/TestResultEventListener.java]: please check for instance type using instanceof

File: core/src/main/java/com/capitalone/dashboard/event/TestResultEventListener.java [hunk 2]
```diff
@@ -0,0 +1,152 @@
+package com.capitalone.dashboard.event;
+
+import com.capitalone.dashboard.model.*;
```
Comment (Sbrenthughes) [inline core/src/main/java/com/capitalone/dashboard/event/TestResultEventListener.java]: Please do not use * import

**Comment by nameisaravind** [comment]
Closing this PR due to additional design changes requested. New PR will be created with all the consolidated changes.

**Pull request closed without merging**